providers/embedded_ansible.md

```diff
@@ -0,0 +1,231 @@
+
+Your ManageIQ DB user (the one in `manageiq/config/database.yml`) must have the `SUPERUSER` privilege (or at least needs to be able to create roles and databases).
+
+
+Config file locations (where to expect it):
+
+  * Debian: `/etc/postgresql/9.6/main/`
+  * Fedora: `/var/lib/pgsql/data/`
+  * MacOSX: `/usr/local/var/postgres/`
+
+Note that these may depend on your version, or oven on how you installed PostgreSQL.
+If you still can't find the right location, you may have luck running `psql -d postgres -c 'show config_file'`.
+
+
+
+Make sure your `postgresql.conf` contains this line:
+
+```
+listen_addresses = '*'
+```
+
+
+Make sure your `pg_hba.conf` contains:
+
+```
+host	all	all	172.17.0.1/24	md5
+```
+
+Note: that `172.17.0.1/24` may depend on the address of your docker network interface - run `ip addr show dev docker0` and you should see an `inet` line with a similar address - use that.
+
+
+Mac users: you may also need to add this one, for `docker-machine`.
+
+```
+host all all 192.168.99.0/24 md5
+```
+
+
+Ensure your DB user (the one in `config/database.yml`) has `superuser` rights..
+
+  * Debian: `sudo su - postgres -c psql -c 'ALTER ROLE "root" SUPERUSER'`
+  * Fedora / MacOSX: `psql -c 'ALTER ROLE "root" SUPERUSER' postgres`
+
+
+### Clean up
+
+If you've already set up an AWX instance this way and want to clean it up:
+
+```
+psql -d postgres -c 'DROP DATABASE awx'
+psql -d postgres -c 'DROP ROLE awx'
+bin/rake evm:db:reset
+bin/rake db:seed
+```
+
+
+If you had previously added an embedded ansible using the [old way](http://talk.manageiq.org/t/howto-setup-embedded-ansible/2291/5?u=himdel), you'll need to clean up the provider (in Rails console):
+
+```
+ManageIQ::Providers::EmbeddedAnsible::Provider.first.destroy!
+```
+
+
+In both cases, you may also need to clean up the old authentications (in Rails console):
+
+```
+db = MiqDatabase.first
+db.authentication_type('ansible_secret_key').delete    # db.ansible_secret_key.delete
+db.ansible_rabbitmq_authentication.delete
+db.ansible_admin_authentication.delete
+db.ansible_database_authentication.delete
+```
+
+
+### Procfiles
+
+Under your `manageiq/` directory, create these 2 files:
+
+`Procfile.ansible`:
+
+```
+ansible: ruby lib/workers/bin/run_single_worker.rb EmbeddedAnsibleWorker
+```
+
+`Procfile.workers`:
+
+```
+generic:                  ruby lib/workers/bin/run_single_worker.rb MiqGenericWorker
+embedded_ansible_refresh: ruby lib/workers/bin/run_single_worker.rb -e 123 ManageIQ::Providers::EmbeddedAnsible::AutomationManager::RefreshWorker
+embedded_ansible_event:   ruby lib/workers/bin/run_single_worker.rb -e 123 ManageIQ::Providers::EmbeddedAnsible::AutomationManager::EventCatcher
+```
+
+In the second file, you'll need to replace that 123 with the id of the newly created **manager** instance.
+
+
+### Setting it up
+
+  * configure your server to enable the ansible role (from Rails console):
```

Comment:
Also, you should be able to `rake evm:start` and turn the role on.  Right @carbonin ?

Reply:
I've never run `evm:start` and don't really intend to. That's simply not how UI does it..

Reply:
@himdel If that's the case, there's some things you just won't be able to do in the UI.  I would not expect all features on the appliance to be feasible or desirable in a rails server process.

Reply:
Do you know which? So far, everything seemed to work...

But, as long as this is mostly around refresh or things like that I think that's perfectly acceptable - at least, I have not needed to `evm:start` for any PR I've reviewed in the last year, so.. :)

At least, still better than that "Embedded ansbile not supported, deal with it" :)

Reply:
@himdel replication, server failover, various platform/provider specific things such as scanning vms/instances on various providers.  We will try to make things easier if we can but at some point we have to say it's too much effort and frankenstein code if there's an easy workaround.

Reply:
@jrafanie Ah, perfect, I don't think we get many UI PRs dealing with those things.

I agree that when you want to test replication, you need to run the whole app.

But this mostly is so that every member of the team can actually test embedded ansible PRs in their devel setups, and for that I think this is OK :).

Reply:
I long ago accepted that there's just some things that are easier to let run on appliances.  Thanks for questioning this because if there are easy changes to do, we should do them.  With that said, though we have to be careful not to add too much hard to maintain hackery.  At some point, we'll need to step back, measure the size of each yak and see if it's worth shaving.

Reply:
Agreed there.. but.. if the UI team can't run code needed to merge PRs, we would literally waste hours on reviews, so... balance :)

Reply:
If you need to run most PRs locally before merge, that sounds like the problem in the workflow.  The only ones I ever run manually on an appliance are ones where tests would be brittle or even useless since they'd require mocks around everything, such as running system commands in ruby and testing some side effect much later (black console partitioning, etc.)